src/main/java/codeu/model/data/MentionList.java

```diff
@@ -0,0 +1,99 @@
+package codeu.model.data;
+
+import codeu.model.store.basic.UserStore;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+public class MentionList {
+    private Message message;
+    private  UserStore userStore;
+
+    public MentionList(Message message, UserStore userStore){
+        message = this.message;
+        userStore = this.userStore;
+    }
+
+    public List<Mention> getList(){
+        List<Mention> mentions = new ArrayList<>();
+        String content = message.getContent();
+
+        Mention mention = null;
+        boolean inMention = false;
+        int startOfMention = -1;
+        int endOfMention = -1;
+
+        for (int i = 0; i < content.length(); i++) {
```

Comment:
I would suggest trying a bunch of different cases in unit tests, in a separate PR.  This stuff is notoriously hard to get right on the first try since it's more than just plumbing (all the persistence stuff is just plumbing: get data from one place and move it to another place without changing it) and edge cases can be quite tricky, as you've seen.  Definitely do it as a separate add-on if you're going to do it, though -- I think this is pretty close to ready for Jorge to integrate.